src/odemis/acq/milling/test/patterns_test.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+"""
+@author: Patrick Cleeve
+
+Copyright © 2024, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import logging
+import unittest
+import json
+import numpy
+from odemis.acq.milling.patterns import RectanglePatternParameters, TrenchPatternParameters, MicroexpansionPatternParameters
+
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)-15s: %(message)s")
+logging.getLogger().setLevel(logging.DEBUG)
+
+class RectanglePatternParametersTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.name = "Rectangle-1"
+        self.width = 10e-6
+        self.height = 10e-6
+        self.depth = 10e-6
+        self.rotation = 0
+        self.center = (0, 0)
+        self.scan_direction = "TopToBottom"
+
+        self.pattern = RectanglePatternParameters(
+            name=self.name,
+            width=self.width,
+            height=self.height,
+            depth=self.depth,
+            rotation=self.rotation,
+            center=self.center,
+            scan_direction=self.scan_direction,
+        )
+
+    def test_assignment(self):
+
+        # test assignment
+        self.assertEqual(self.pattern.name.value, self.name)
+        self.assertEqual(self.pattern.width.value, self.width)
+        self.assertEqual(self.pattern.height.value, self.height)
+        self.assertEqual(self.pattern.depth.value, self.depth)
+        self.assertEqual(self.pattern.rotation.value, self.rotation)
+        self.assertEqual(self.pattern.center.value, self.center)
+        self.assertEqual(self.pattern.scan_direction.value, self.scan_direction)
+
+    def test_json(self):
```

Comment:
Can you rename this and the function below to `test_dict` or `test_to_from_dict`, and also rename `rectangle_pattern_json` to `rectangle_pattern_dict` etc.